content/advent-2017/building-medical-reports-in-go.md

```diff
@@ -0,0 +1,241 @@
+	pdf.AliasNbPages("")
+	pdf.SetHeaderFunc(report.headerFunc)
+	pdf.SetFooterFunc(report.footerFunc)
+
+	report.document = pdf
+	return report
+}
+```
+
+Our struct `Report` just holds the configured document as you can see in the last line before actually return it.
+
+The ideia here is just to show how simple is to create your documents using this library. You could check the [docs](https://godoc.org/github.com/jung-kurt/gofpdf) to understand what each function does. Let's move one step further.
+
+The main code lives in a `HandleFunc`, which was created just to be able to see this PDF usign a browser instead of generate it in the file system.
+
+```go
+http.HandleFunc("/report", func(w http.ResponseWriter, r *http.Request) {
+    report := NewReport()
+    report.PatientHeader()
+    report.Diagnostic()
+    report.GeneList()
+    report.TechnicalResponsible()
+    report.Method()
+    report.QualityFlags()
+    report.VUS()
+    report.Comments()
+
+
+    if err := report.document.Output(w); err != nil {
+        log.Print(err)
+    }
+})
+```
+
+Each function in the `report` object (created by our `NewReport` as we saw above) is a section builder for that given name. We build the patient header, with information like name, age, etc, diagnostic section which is where we explain if the patient have a positive or negative result for that specific exam and so on.
+
+Let's get the `Method` function to see how it works under the hood.
+
+```go
+func (r *Report) Method() {
+	fs := 10.0
+	r.document.SetFontSize(fs)
+
+	content := "Captura de exons com Nextera Exome Capture seguida por sequenciamento de nova " +
+		"geração com Illumina HiSeq. Alinhamento e identificação de variantes utilizando protocolos " +
+		"de bioinformática, tendo como referência a versão GRCh37 do genoma humano. Análise médica " +
+		"orientada pelas informações que motivaram a realização deste exame."
+
+	r.drawLine(fs, "<b>Método</b>", 10)
+
+	// MultiCell(width, height, content, border, align, fill)
+	r.document.MultiCell(0, 5, r.encodingFunc(content), "", "", false)
+	r.lineStroke()
+}
+```
+
+We just set the font size for that section, build a text for it and add it to the document using the `MultiCell` function. Easy! No x/y alignment, text limit, etc. The library will handle it based on our previous configuration at `NewReport`.
+
+Going back to the main function, at the end we write the final document back to the `ResposeWriter` and we're done!
+
+Below you could see the result of this POC: a report 99% similar to what we have in python in just 351 LOC:
+
+![](/postimages/advent-2017/building-medial-reports-in-go/poc-page-1.png)
+![](/postimages/advent-2017/building-medial-reports-in-go/poc-page-2.png)
+
+## From Python to Go
+
+As we saw previously, we need to change how Python and Go communicate each other. Instead send an `application/pdf` as before, now the python API need to serve all the data needed to construct the report on the Go side.
```

Comment:
Instead of `communicate each other`, try `communicate with each other`.  And instead of `need to serve`, try `needs to serve`.